### PR TITLE
Feature/ferment api service

### PR DESCRIPTION
**Pull request by brentkinney**

### CODE REVIEW

File: SabreSprings.Brewing.Services/FermentabuoyLogService.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        /// <summary>
+        /// This method takes the dto from the API controller and translates the data to a 
+        /// matching entity which is then given to the data provider to be inserted into the database.
+        /// </summary>
+        /// <param name="fermentabuoyLogDto"> The Dto received by the controller layer</param>
+        /// <returns></returns>
+        public async Task<FermentabuoyLogDto> AddFermentabuoyLog(FermentabuoyLogDto fermentabuoyLogDto) 
+        {            
+            
+            FermentabuoyLog Log = new FermentabuoyLog()
```
Comment (bkinney91) [inline SabreSprings.Brewing.Services/FermentabuoyLogService.cs]: Local variables should be camel case.

File: SabreSprings.Brewing.Data/FermentabuoyLogDataProvider.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        }
+
+        /// <summary>
+        /// This method receives a FermentationLog entity from the servcie and inserts the data into the SQLite DB
+        /// </summary>
+        /// <param name="log"> The Entity that is received from the service</param>
+        /// <returns></returns>
+        public async Task<FermentabuoyLog> AddFermentabuoyLog(FermentabuoyLog log) 
+        {            
+            string sql = "Insert into FermentationLog (Name, Temperature, Gravity, Angle, DeviceNumber, Battery, RSSI) VALUES (@Name, @Temperature, @Gravity, @Angle, @DeviceId, @Battery, @RSSI);";
```
Comment (bkinney91) [inline SabreSprings.Brewing.Data/FermentabuoyLogDataProvider.cs]: DeviceNumber and @DeviceID should match, which is correct?

File: SabreSprings.Brewing.Database/FermentationLog.sql [hunk 3]
```diff
@@ -1,11 +1,12 @@
 ﻿CREATE TABLE [dbo].[FermentationLog]
```
Comment (bkinney91) [inline SabreSprings.Brewing.Database/FermentationLog.sql]: [dbo] is SQL Server, not sqlite

File: SabreSprings.Brewing.Database/FermentationLog.sql [hunk 4]
```diff
@@ -1,11 +1,12 @@
...
+	[Name] TEXT NOT NULL,
 	[Temperature] DECIMAL NOT NULL,
 	[Gravity] DECIMAL NOT NULL,
 	[Angle] DECIMAL NOT NULL,
 	[DeviceNumber] INTEGER NOT NULL,
-	[Created] DateTime default CURRENT_TIMESTAMP,
-	FOREIGN KEY(Batch) REFERENCES Batches(Id),
+	[Battery] DECIMAL NOT NULL,
+	[RSSI] INTEGER NOT NULL,
+	[Created] DateTime default CURRENT_TIMESTAMP,
```
Comment (bkinney91) [inline SabreSprings.Brewing.Database/FermentationLog.sql]: Last item should not have a comma

File: SabreSprings.Brewing.Models/DataTransfer/FermentabuoyLogDto.cs [hunk 5]
```diff
@@ -7,10 +7,10 @@ namespace SabreSprings.Brewing.Models.DataTransfer
     public class FermentabuoyLogDto
     {
         public string Name { get; set; }
-        public int ID { get; set; }
+        public int DeviceId { get; set; }
```
Comment (bkinney91) [inline SabreSprings.Brewing.Models/DataTransfer/FermentabuoyLogDto.cs]: This needs to be ID

File: SabreSprings.Brewing.Services/FermentabuoyLogService.cs [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+        /// </summary>
+        /// <param name="fermentabuoyLogDto"> The Dto received by the controller layer</param>
+        /// <returns></returns>
+        public async Task<FermentabuoyLogDto> AddFermentabuoyLog(FermentabuoyLogDto fermentabuoyLogDto) 
+        {            
+            
+            FermentabuoyLog log = new FermentabuoyLog()            
+            {
+                Name = fermentabuoyLogDto.Name,
+                DeviceId = fermentabuoyLogDto.DeviceId,
```
Comment (bkinney91) [inline SabreSprings.Brewing.Services/FermentabuoyLogService.cs]: DeviceID = fermentabuoyLogDto.ID

File: SabreSprings.Brewing.Database/FermentationLog.sql [hunk 7]
```diff
@@ -1,11 +1,12 @@
 ﻿CREATE TABLE [dbo].[FermentationLog]
 (
 	[Id] INT NOT NULL PRIMARY KEY,
-	[Batch] INTEGER NOT NULL,
+	[Name] TEXT NOT NULL,
 	[Temperature] DECIMAL NOT NULL,
 	[Gravity] DECIMAL NOT NULL,
 	[Angle] DECIMAL NOT NULL,
 	[DeviceNumber] INTEGER NOT NULL,
```
Comment (bkinney91) [inline SabreSprings.Brewing.Database/FermentationLog.sql]: This does not match your entity....

File: SabreSprings.Brewing.Data/FermentabuoyLogDataProvider.cs [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+            using (IDbConnection db = new SqliteConnection(_configuration.GetConnectionString("SabreSpringsBrewing")))
+            {
+                log = await db.QueryFirstAsync<FermentabuoyLog>(sql);
```
Comment (bkinney91) [inline SabreSprings.Brewing.Data/FermentabuoyLogDataProvider.cs]: Inserting is not _querying_ use "ExecuteAsync" instead.

File: SabreSprings.Brewing.Services/FermentabuoyLogService.cs [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+                Temperature = fermentabuoyLogDto.Temperature,
+                Battery = fermentabuoyLogDto.Battery,
+                Gravity = fermentabuoyLogDto.Gravity,
+                RSSI = fermentabuoyLogDto.RSSI
+            };
+            await FermentabuoyLogDataProvider.AddFermentabuoyLog(log);
+            return Ok(log);
+        }
+
+        private FermentabuoyLogDto Ok(FermentabuoyLog log)
```
Comment (bkinney91) [inline SabreSprings.Brewing.Services/FermentabuoyLogService.cs]: Remove this

File: SabreSprings.Brewing.Services/FermentabuoyLogService.cs [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+                Name = fermentabuoyLogDto.Name,
+                DeviceId = fermentabuoyLogDto.DeviceId,
+                Angle = fermentabuoyLogDto.Angle,
+                Temperature = fermentabuoyLogDto.Temperature,
+                Battery = fermentabuoyLogDto.Battery,
+                Gravity = fermentabuoyLogDto.Gravity,
+                RSSI = fermentabuoyLogDto.RSSI
+            };
+            await FermentabuoyLogDataProvider.AddFermentabuoyLog(log);
+            return Ok(log);
```
Comment (bkinney91) [inline SabreSprings.Brewing.Services/FermentabuoyLogService.cs]: Why are you calling the Ok function?

Comment (bkinney91) [reply SabreSprings.Brewing.Services/FermentabuoyLogService.cs]: This add should not not return anything

File: SabreSprings.Brewing.Services/FermentabuoyLogService.cs (hunk 1 above)
Comment (brentkinney) [reply SabreSprings.Brewing.Services/FermentabuoyLogService.cs]: fixed

File: SabreSprings.Brewing.Data/FermentabuoyLogDataProvider.cs (hunk 2 above)
Comment (brentkinney) [reply SabreSprings.Brewing.Data/FermentabuoyLogDataProvider.cs]: SQL table changed to DeviceId

File: SabreSprings.Brewing.Database/FermentationLog.sql (hunk 3 above)
Comment (brentkinney) [reply SabreSprings.Brewing.Database/FermentationLog.sql]: Create table query changed to match SQLite

File: SabreSprings.Brewing.Database/FermentationLog.sql (hunk 4 above)
Comment (brentkinney) [reply SabreSprings.Brewing.Database/FermentationLog.sql]: Comma removed

File: SabreSprings.Brewing.Models/DataTransfer/FermentabuoyLogDto.cs (hunk 5 above)
Comment (brentkinney) [reply SabreSprings.Brewing.Models/DataTransfer/FermentabuoyLogDto.cs]: Dto was changed back to ID

File: SabreSprings.Brewing.Services/FermentabuoyLogService.cs (hunk 6 above)
Comment (brentkinney) [reply SabreSprings.Brewing.Services/FermentabuoyLogService.cs]: Service translation was changed from incorrect DeviceId to correct ID

File: SabreSprings.Brewing.Database/FermentationLog.sql (hunk 7 above)
Comment (brentkinney) [reply SabreSprings.Brewing.Database/FermentationLog.sql]: Fixed in previous table comment

File: SabreSprings.Brewing.Data/FermentabuoyLogDataProvider.cs (hunk 8 above)
Comment (brentkinney) [reply SabreSprings.Brewing.Data/FermentabuoyLogDataProvider.cs]: corrected method execution

File: SabreSprings.Brewing.Services/FermentabuoyLogService.cs (hunk 9 above)
Comment (brentkinney) [reply SabreSprings.Brewing.Services/FermentabuoyLogService.cs]: removed and method fixed to not return anything

File: SabreSprings.Brewing.Services/FermentabuoyLogService.cs (hunk 10 above)
Comment (brentkinney) [reply SabreSprings.Brewing.Services/FermentabuoyLogService.cs]: method has been fixed to not return anything.

**Pull request merged**